dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-trino/src/main/java/org/apache/dolphinscheduler/plugin/datasource/trino/param/TrinoDataSourceProcessor.java

```diff
@@ -0,0 +1,150 @@
+    }
+
+    @Override
+    public BaseDataSourceParamDTO createDatasourceParamDTO(String connectionJson) {
+        TrinoConnectionParam connectionParams = (TrinoConnectionParam) createConnectionParams(connectionJson);
+
+        String[] hostSeperator = connectionParams.getAddress().split(Constants.DOUBLE_SLASH);
+        String[] hostPortArray = hostSeperator[hostSeperator.length - 1].split(Constants.COMMA);
+
+        TrinoDataSourceParamDTO TrinoDatasourceParamDTO = new TrinoDataSourceParamDTO();
+        TrinoDatasourceParamDTO.setPort(Integer.parseInt(hostPortArray[0].split(Constants.COLON)[1]));
+        TrinoDatasourceParamDTO.setHost(hostPortArray[0].split(Constants.COLON)[0]);
+        TrinoDatasourceParamDTO.setDatabase(connectionParams.getDatabase());
+        TrinoDatasourceParamDTO.setUserName(connectionParams.getUser());
+        TrinoDatasourceParamDTO.setOther(parseOther(connectionParams.getOther()));
+
+        return TrinoDatasourceParamDTO;
+    }
+
+    @Override
+    public BaseConnectionParam createConnectionParams(BaseDataSourceParamDTO datasourceParam) {
+        TrinoDataSourceParamDTO TrinoParam = (TrinoDataSourceParamDTO) datasourceParam;
+        String address =
+                String.format("%s%s:%s", DataSourceConstants.JDBC_TRINO, TrinoParam.getHost(), TrinoParam.getPort());
+        String jdbcUrl = address + "/" + TrinoParam.getDatabase();
+
+        TrinoConnectionParam TrinoConnectionParam = new TrinoConnectionParam();
+        TrinoConnectionParam.setUser(TrinoParam.getUserName());
+        TrinoConnectionParam.setPassword(PasswordUtils.encodePassword(TrinoParam.getPassword()));
+        TrinoConnectionParam.setOther(transformOther(TrinoParam.getOther()));
+        TrinoConnectionParam.setAddress(address);
+        TrinoConnectionParam.setJdbcUrl(jdbcUrl);
+        TrinoConnectionParam.setDatabase(TrinoParam.getDatabase());
+        TrinoConnectionParam.setDriverClassName(getDatasourceDriver());
+        TrinoConnectionParam.setValidationQuery(getValidationQuery());
+        TrinoConnectionParam.setProps(TrinoParam.getOther());
+
+        return TrinoConnectionParam;
+    }
+
+    @Override
+    public ConnectionParam createConnectionParams(String connectionJson) {
+        return JSONUtils.parseObject(connectionJson, TrinoConnectionParam.class);
+    }
+
+    @Override
+    public String getDatasourceDriver() {
+        return DataSourceConstants.COM_TRINO_JDBC_DRIVER;
+    }
+
+    @Override
+    public String getValidationQuery() {
+        return DataSourceConstants.TRINO_VALIDATION_QUERY;
+    }
+
+    @Override
+    public String getJdbcUrl(ConnectionParam connectionParam) {
+        TrinoConnectionParam TrinoConnectionParam = (TrinoConnectionParam) connectionParam;
+        if (!StringUtils.isEmpty(TrinoConnectionParam.getOther())) {
+            return String.format("%s?%s", TrinoConnectionParam.getJdbcUrl(), TrinoConnectionParam.getOther());
```

Comment:
may `StringUtils.isNotEmpty()` should be better, WDYT

Reply:
done.